### PR TITLE
Feat:Added bot detection and disabled analytics if a bot

### LINKED ISSUE #226: opt bots out of analytics
## Problem Statement

[Basecamp issue](https://3.basecamp.com/3926363/buckets/27088350/card_tables/cards/7438138554)

Bots are adding to our analytics data and they shouldn't be

## Resolution Criteria

- [ ] Bots are not adding to our analytics data

## Context

You should be able to use this library: https://www.npmjs.com/package/isbot

### PR DESCRIPTION
**Pull request by rajitharussel**
## 🐛 Issue

Closes #226 

## ✏️ Solution

implemented way to identify bots and disbled logs.

## 🔬 To Test

1. I build this snippet to load the page as a bot. and added some logs here so we can check if analytics are executed. 

```
const puppeteer = require('puppeteer');

async function scrapeData(url) {
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    // Listen for all console events and log them
    page.on('console', msg => {
        console.log('CONSOLE:', msg.text());
    });
    await page.goto(url, { waitUntil: 'networkidle0' });  // Ensures the page has loaded completely

    // Extract data
    const data = await page.evaluate(() => {
        const element = document.querySelector('.apollos-widget');
        return element ? element.innerText : null;
    });

    await browser.close();
    return data;
}

scrapeData('http://localhost:3000/ac/the-rest-of-god-TWVkaWFDb250ZW50SXRlbS02MGQ0Njg0YS03NWQ2LTQ2MzAtYTJjOC1iMDI0MWZkMTE5NzQ=').then(console.log).catch(console.error);

```

to see if analytics executed add followings here

https://github.com/ApollosProject/apollos-embeds/blob/5adf7e6b1d4d8efb8553a2320e0331350b348549/packages/web-shared/analytics/amplitude.js#L3

Screenshot 2024-07-04 at 17 43 14

2. or can use analytics realtime dashboard directly

#### Before
Screenshot 2024-07-04 at 17 47 42

#### After
Screenshot 2024-07-04 at 17 48 17

### CODE REVIEW

File: micro-service/src/utils/index.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export const getChurchSlug = (host) => {
+  return 'king_of_kings'
```
Comment (redreceipt) [inline micro-service/src/utils/index.js:2]:
```suggestion
```

**Review by redreceipt — CHANGES_REQUESTED** [review]

File: micro-service/src/utils/index.js (hunk 1 above)
Comment (rajitharussel) [reply micro-service/src/utils/index.js:2]: oh thats a mistake

File: packages/web-shared/providers/AnalyticsProvider.js [hunk 2]
```diff
@@ -55,16 +56,22 @@ export const GET_ANALYTICS_USER = gql`
 export const AnalyticsProvider = ({ children, church }) => {
   const { currentChurch } = useCurrentChurch();
   const { currentUser } = useCurrentUser();
+  const isBot = useIsBot();
```
Comment (vinnyjth) [inline packages/web-shared/providers/AnalyticsProvider.js:59]: Does this need to be a hook, the call looks synchronous :)

Comment (redreceipt) [reply packages/web-shared/providers/AnalyticsProvider.js:59]: I agree, don't really need a hook

Comment (rajitharussel) [reply packages/web-shared/providers/AnalyticsProvider.js]: this is fixed

**Review by redreceipt — APPROVED** [review]

**Pull request merged**